### DISCUSSION TITLE
CVE-2022-10735

### DISCUSSION THREAD

**Discussion by Fidanze** [RustPython/RustPython#4164] [General]
https://github.com/python/cpython/issues/96834
Is it work on this realization?
if yes => How you will solve problem? like CPython core developers?

**Comment by youknowone** [comment]
Comparing to CPython, RustPython has a lot more attackable surface of that kind of attacks due to worse optimizations for here and there.
I don't think we have motivation to fix that specific problem while we have a lot more similar problems. They will update the performance in next(or the next of next) release and we can catch it up later.

**Comment by fanninpm** [comment]
If I recall correctly, we rely on the [`num-bigint`](https://lib.rs/crates/num-bigint) crate to handle Python's `int` type. It might be worth asking the maintainers of that crate if it is susceptible to the kind of attack described in CVE-2022-10735.
